client/components/close-on-escape/test/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import ReactDom, { unmountComponentAtNode } from 'react-dom';
+import { spy } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import CloseOnEscape from '../';
+
+const simulateEscapeKeydown = () => (
```

Comment:
Yes, that's true. The most advanced and also the slowest test suite is the one @gwwar has pointed out :)
Definitely using `enzyme` is the way to go. [Here](https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/simulate.md) you can check actual API of `simulate` method executed on mounted element. In case of `keydown` event we probably can dispatch it also on document level.